java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -385,10 +392,66 @@ private API getXPAPI(
     return API.UNKNOWN;
   }
 
-  private boolean isArgumentMatchesFlagName(ExpressionTree argTree, Symbol argSym) {
+  private boolean isArgumentMatchesFlagName(
+      ExpressionTree argTree, VisitorState state, Symbol argSym) {
     return (isLiteralTreeAndMatchesFlagName(argTree)
         || isVarSymbolAndMatchesFlagName(argSym)
-        || isSymbolAndMatchesFlagName(argSym));
+        || isSymbolAndMatchesFlagName(argSym)
+        || isMatchingEnumFieldValue(argTree, state));
+  }
+
+  private boolean isMatchingEnumFieldValue(ExpressionTree argTree, VisitorState state) {
+    if (argTree.getKind() != Kind.METHOD_INVOCATION) {
```

Comment:
Here is another proposal. I believe the enum declarations will be processed by ErrorProne before the invocations are processed. Is it possible to infer the enum flag declarations and methods when the enum class is parsed, store the results and reuse it in `getXPAPI`? 

This will be a problem if the enum declarations are in a different compilation unit (or in a library jar) and are not part of the current compilation. To overcome this, we can have an option to check for the stored results and if unavailable, perform the computation that you have currently, and this option can be turned off by default. 

I think this inference and use will suffice in a majority of the cases.

Reply:
Isn't this where two passes are needed? Because we can't rely on the enum class being parsed before its usages?

Although I am happy to cache results for multiple reasons.